cmd/snap-bootstrap/cmd_initramfs_mounts.go

```diff
@@ -298,7 +298,60 @@ func generateMountsModeRun() error {
 		kernelPath := filepath.Join(dataDir, "system-data", dirs.SnapBlobDir, kernel.Filename())
 		fmt.Fprintf(stdout, "%s %s\n", kernelPath, filepath.Join(runMnt, "kernel"))
 	}
-	// 3.1 Write the modeenv out again
+
+	// 3.1 Maybe mount the snapd snap on first boot of run-mode
+	// TODO:UC20: Make RecoverySystem empty after successful first boot
+	// somewhere in devicestate
+	if modeEnv.RecoverySystem != "" {
+		isSnapdMounted, err := osutilIsMounted(filepath.Join(runMnt, "snapd"))
+		if err != nil {
+			return err
+		}
+
+		if !isSnapdMounted {
+			// load the recovery system  and generate mounts for kernel/base
+			systemSeed, err := seed.Open(seedDir, modeEnv.RecoverySystem)
+			if err != nil {
+				return fmt.Errorf("cannot open seed: %v", err)
```

Comment:
the other similar code doesn't rewrap errors, we should be consistent

Reply:
Undone, no longer wraps